examples/multi-gpu-movielens/01-03b-MultiGPU-Download-Convert-ETL-with-NVTabular-Training-with-PyTorch.ipynb

```diff
@@ -0,0 +1,683 @@
+    "        pool_allocator=True,\n",
+    "        initial_pool_size=None, # Use default size\n",
+    "    )\n",
+    "    \n",
+    "client.run(_rmm_pool)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "68600ce4",
+   "metadata": {},
+   "source": [
+    "#### Defining our Preprocessing Pipeline\n",
+    "\n",
+    "This subsection is based on [getting-started-movielens/02-ETL-with-NVTabular.ipynb](../getting-started-movielens/02-ETL-with-NVTabular.ipynb). The only difference is that we initialize the NVTabular workflow using the LocalCUDACluster client with `nvt.Workflow(output, client=client)`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "e490f957",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/conda/envs/merlin/lib/python3.8/site-packages/distributed/worker.py:3560: UserWarning: Large object of size 1.90 MiB detected in task graph: \n",
+      "  (\"('read-parquet-9a52beaf761dbfd2bf91886e0b402bc6' ... 1886e0b402bc6')\n",
+      "Consider scattering large objects ahead of time\n",
+      "with client.scatter to reduce scheduler burden and \n",
+      "keep data on workers\n",
+      "\n",
+      "    future = client.submit(func, big_data)    # bad\n",
+      "\n",
+      "    big_future = client.scatter(big_data)     # good\n",
+      "    future = client.submit(func, big_future)  # good\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "movies= cudf.read_parquet(pathlib.Path(BASE_DIR, \"ml-25m\", \"movies_converted.parquet\"))\n",
+    "joined = (\n",
+    "    ['userId', 'movieId'] >> \n",
+    "    nvt.ops.JoinExternal(movies, on=['movieId'])\n",
+    ")\n",
+    "cat_features = joined >> nvt.ops.Categorify()\n",
+    "ratings = nvt.ColumnGroup(['rating']) >> (lambda col: (col>3).astype('int8'))\n",
+    "output = cat_features+ratings\n",
+    "# USE client in NVTabular workfow\n",
+    "workflow = nvt.Workflow(output, client=client)\n",
+    "!rm -rf $BASE_DIR/train\n",
+    "!rm -rf $BASE_DIR/valid\n",
+    "train_iter = nvt.Dataset([str(pathlib.Path(BASE_DIR, \"train.parquet\"))], part_size=\"100MB\")\n",
+    "valid_iter = nvt.Dataset([str(pathlib.Path(BASE_DIR, \"valid.parquet\"))], part_size=\"100MB\")\n",
+    "workflow.fit(train_iter)\n",
+    "workflow.save(pathlib.Path(BASE_DIR, \"workflow\"))\n",
+    "shuffle = Shuffle.PER_WORKER  # Shuffle algorithm\n",
+    "out_files_per_proc = 4        # Number of output files per worker\n",
+    "workflow.transform(train_iter).to_parquet(\n",
+    "    output_path=pathlib.Path(BASE_DIR, \"train\"),\n",
+    "    shuffle=shuffle,\n",
+    "    out_files_per_proc=out_files_per_proc,\n",
+    ")\n",
+    "workflow.transform(valid_iter).to_parquet(\n",
+    "    output_path=pathlib.Path(BASE_DIR, \"valid\"),\n",
+    "    shuffle=shuffle,\n",
+    "    out_files_per_proc=out_files_per_proc,\n",
+    ")\n",
+    "\n",
+    "client.shutdown()"
```

Comment:
Should we also close the cluster? (like `cluster.close()` here)?

Reply:
Yes, probably so. Weirdly, that method is not in the Dask Distributed API documentation.